lib/services/plugin-variables-service.ts

```diff
@@ -0,0 +1,102 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import * as helpers from "./../common/helpers";
+
+export class PluginVariablesService implements IPluginVariablesService {
+	private static PLUGIN_VARIABLES_KEY = "variables";
+
+	constructor(private $errors: IErrors,
+		private $pluginVariablesHelper: IPluginVariablesHelper,
+		private $projectData: IProjectData,
+		private $projectDataService: IProjectDataService,
+		private $prompter: IPrompter) { }
+
+	public getPluginVariablePropertyName(pluginData: IPluginData): string {
+		return `${pluginData.name}-${PluginVariablesService.PLUGIN_VARIABLES_KEY}`;
+	}
+
+	public savePluginVariablesInProjectFile(pluginData: IPluginData): IFuture<void> {
+		return (() => {
+			let values = Object.create(null);
+			this.executeForAllPluginVariables(pluginData, (pluginVariableData: IPluginVariableData) =>
+				(() => {
+					let pluginVariableValue = this.getPluginVariableValue(pluginVariableData).wait();
+					if(!pluginVariableValue) {
```

Comment:
This could be extracted into a dedicated function `ensurePluginvarValue`, the same code is used in other places as well.